src/components/Header/Header.jsx

```diff
@@ -139,105 +144,130 @@ const SettingsGroup = ({ children }) => {
-    );
-
-    const LogOut = () => (
-      <BugLink
-        href={`${process.env.REACT_APP_API_HOST}/logout`}
-        rel="noopener noreferrer"
-      >
-        <LogOutButton />
-        <span>
-          {strings.app_logout}
-        </span>
-      </BugLink>
-    );
-
-    const { Announce } = this.state;
-
-    return (
-      <div>
-        <ToolbarHeader>
-          <VerticalAlignDiv>
-            <LogoGroup small={small} />
-            {small && <LinkGroup navbarPages={navbarPages} />}
-          </VerticalAlignDiv>
-          {!disableSearch && <SearchGroup />}
-          <VerticalAlignDiv style={{ marginLeft: '16px' }}>
-            {small && <AccountGroup />}
-            <SettingsGroup>
-              <LocalizationMenu />
-              <ReportBug />
-              {user ? <LogOut /> : null}
-            </SettingsGroup>
-          </VerticalAlignDiv>
-        </ToolbarHeader>
-        { location.pathname !== '/' && Announce && <Announce /> }
-      </div>
-    );
-  }
-}
+const LogoGroup = ({ onMenuClick }) => (
+  <div style={{ marginRight: 16 }}>
+    <VerticalAlignToolbar>
+      <MenuButtonWrapper>
+        <IconButton edge="start" color="inherit" onClick={onMenuClick}>
+          <MenuIcon />
+        </IconButton>
+      </MenuButtonWrapper>
+      <AppLogoWrapper>
+        <AppLogo />
+      </AppLogoWrapper>
+    </VerticalAlignToolbar>
+  </div>
+);
+
+LogoGroup.propTypes = {
+  onMenuClick: PropTypes.func,
+};
+
+const SearchGroup = () => (
+  <VerticalAlignToolbar style={{ marginLeft: 'auto' }}>
+    <ActionSearch style={{ marginRight: 6, opacity: '.6' }} />
+    <SearchForm />
+  </VerticalAlignToolbar>
+);
+
+const AccountGroup = () => (
+  <VerticalAlignToolbar>
+    <AccountWidget />
+  </VerticalAlignToolbar>
+);
+
+const ReportBug = ({ strings }) => (
+  <DropdownMenuItem component="a" href={REPORT_BUG_PATH} target="_blank" rel="noopener noreferrer" >
+    <BugReport style={{ marginRight: 32, width: 24, height: 24 }} />
+    {strings.app_report_bug}
+  </DropdownMenuItem>
+);
+
+ReportBug.propTypes = {
+  strings: PropTypes.shape({}),
+};
+
+const LogOut = ({ strings }) => (
+  <BugLink
+    href={`${process.env.REACT_APP_API_HOST}/logout`}
+    rel="noopener noreferrer"
+  >
+    <LogOutButton />
+    <span>
+      {strings.app_logout}
+    </span>
+  </BugLink>
+);
 
-const mapStateToProps = state => ({
-  small: state.browser.greaterThan.small,
-  user: state.app.metadata.data.user,
-  strings: state.app.strings,
-});
+LogOut.propTypes = {
+  strings: PropTypes.shape({}),
+};
+
+const Header = ({
+  location, disableSearch,
+}) => {
+  const [Announce, setAnnounce] = useState(null);
+  const [menuIsOpen, setMenuState] = useState(false);
+  const navigationState = useContext(navigationContext);
+  const small = useSelector(state => state.browser.greaterThan.small);
+  const user = useSelector(state => state.app.metadata.data.user);
+  const strings = useSelector(state => state.app.strings);
```

Comment:
So just for clarity for people who are not used to write components with hooks yet I've decided to add an explanation here on what is going on.

```js
// Here I create a state for the Announce component
// It's the equivalent to this.state.Announce and this.setState({ Announce })
const [Announce, setAnnounce] = useState(null);

// Here I create a second state for the menu drawer state (we can have multiple states with hooks)
// This is the equivalent to this.state.menuIsOpen and this.setState({ menuIsOpen })
const [menuIsOpen, setMenuState] = useState(false);
```

// Here I use a context to pass down navigation pages from the App.jsx
// The useContext hook is a hook coming from React to access data from a context without
// the use of any HoC
// inside we have navigationState.navPages and navigationState.drawerPages
const navigationState = useContext(navigationContext);

// Here I get the small value from redux
// useSelector is a hook from react-redux which allows us to select any value from the current redux state
const small = useSelector(state => state.browser.greaterThan.small);

// Same here, this gets the current user from redux
const user = useSelector(state => state.app.metadata.data.user);

// Same here, we get the apps strings
const strings = useSelector(state => state.app.strings);